### PR TITLE
feat(event-stats): Drop equation field validation for topEvent queries

### PR DESCRIPTION
**Pull request by narsaynorath**
The reason for the validation was to encourage the user
to add all fields in Discover to get auto-complete, but we are bypassing
this validation to support sorting by arbitrary custom equations in the
widget builder.

This change was approved in the API pain points discussion, see [this notion doc](https://www.notion.so/sentry/Meeting-API-Pain-Points-2022-04-12-dfdf5bbe67d14f089af802f176fdf68d#b2862a9888a948ff8b29ecdd0fc9ceef)

### CODE REVIEW

File: src/sentry/snuba/discover.py [hunk 1]
```diff
@@ -209,6 +209,7 @@ def query(
     referrer=None,
     auto_fields=False,
     auto_aggregations=False,
+    auto_equation_fields=False,
```
Comment (shruthilayaj) [inline src/sentry/snuba/discover.py:212]: can we rename this to something more descriptive like "include_equation_fields" or "add_equation_fields" or something?

File: src/sentry/snuba/discover.py [hunk 2]
```diff
@@ -726,6 +735,7 @@ def top_events_timeseries(
                 auto_aggregations=True,
                 use_aggregate_conditions=True,
                 use_snql=use_snql,
+                auto_equation_fields=True,
```
Comment (shruthilayaj) [inline src/sentry/snuba/discover.py:738]: should we add a test case in test_discover.py to cover this change?

**Review by shruthilayaj — APPROVED** [review]

File: src/sentry/snuba/discover.py (hunk 1 above)
Comment (narsaynorath) [reply src/sentry/snuba/discover.py:212]: I like `include_equation_fields` over `add_equation_fields`, `include` makes me feel more like something is happening for me and reminds me of include filters for URL endpoints. `add` feels different

File: src/sentry/snuba/discover.py (hunk 2 above)
Comment (narsaynorath) [reply src/sentry/snuba/discover.py]: I added a test to assert that this is called with `include_equation_fields=True`. I'm not sure if that's what you had in mind but we can follow up tomorrow if not and I can write a proper test.

**Review by dashed — APPROVED** [review]

**Pull request merged**